### ISSUE TITLE
Bible Version Changes

### ISSUE THREAD

**Issue by amallison** [OliveTreeBible/OliveTreeUrlExample#3]
I use both the ESV & NIV. I usually use NIV. I also use the ReadingPlan app to record readings. What determines which Bible version the URL displays? It looks like it sometime (maybe after a day) not all the time the Bible+ prefers to open the ESV.  I'd like the URL to allow you to also specify the Bible version. Anyway could you make it just use the last version viewed in Bible+. I find this problematic because if I do a partial verse highlight in ESV it isn't seen in NIV.
Thanks

**Comment by tomhamming** [comment]
We don't currently support specifying Bible versions with the URL. But you should be able to change what opens up by changing the Default Bible for Verse Hyperlinks setting in App Settings under Hyperlink Settings.

**Comment by theipadforpastors** [comment]
Tom, is there a possibility you'll be adding that functionality to the URL scheme at some point?

**Comment by tomhamming** [comment]
@theipadforpastors it's possible, but not in the plans at the moment. What's your use case?

**Comment by theipadforpastors** [comment]
Well, I use the KJV and ESV at different times. It would be nice to be able to specify when I use a LCP action.
However, being able to start a search from a URL action is a much, much higher request:-)

**Comment by tomhamming** [comment]
@theipadforpastors what's the use case for a search URL?

**Comment by theipadforpastors** [comment]
It would allow me to open a search from Drafts.app, Workflow...really anywhere.
Basically, I could be working on my message in Pages, select a word, tap share and then from there open a search directly in Olive Tree.
This would be quickest from Drafts or Editorial. I could literally open a search from the keyboard without ever touching the screen at all.

**Comment by tomhamming** [comment]
@theipadforpastors having URL support for something doesn't mean we'll show up in a share sheet. That would require an app extension. But you could generate a link in Pages and then click on it, I suppose.

**Comment by theipadforpastors** [comment]
Yes, I understand that. And while I like Extension support for some things, for text I actually prefer URL.
And I wouldn't be generating a URL to open search from Pages, what I would do is use Workflow to wrap text received from the share menu in a URL which it would pass on to Olive Tree. That's cool and geeky, but it's not actually where I really want to use URL search support from.
Drafts and Launch Center Pro would make URL support from Olive Tree so awesome. Open a Draft, type a search word and tap an action and, bam, it opens right to Olive Tree with search already populated.
And with Pythonista, someone could literally build their own custom UI for Greek search and then send the search to Olive Tree. The possibilities are almost limitless.
While we're at it, I'd also love to have the same thing for Look Up. I *love* the Look Up function, but it's a bit buried. Exposing it to a URL would be awesome.

**Comment by tomhamming** [comment]
@theipadforpastors cool, just making sure we're on the same page. I'll look into this.

**Comment by theipadforpastors** [comment]
Really, really appreciate that.
Y'all make the _best_ mobile Bible Study app, and (for my uses) this would make it even better. :-)

**Comment by tomhamming** [comment]
@theipadforpastors as of version 6.2.5:

`olivetree://search/<search term>/<product ID>`
Or:
`olivetree://search/<search term>/<title or abbreviated title of resource>`

Search term and title/abbreviated title should be URL-encoded if necessary. If you don't specify a product (via an ID or title) we search the Default Bible for Verse Hyperlinks.

Also:
`olivetree://lookup/<search term>`
Where, again, search term is URL-encoded.

Let me know how that works.

**Comment by theipadforpastors** [comment]
Wow! Just wow! Like Christmas in March. Thanks so much. I'll be working on writing a Drafts.app action and LCP action and sharing them. Also, I'll put a pass on the news to a few friends with Pocasts and a Blog hoping they'll expose this to a wider audience.

**Comment by theipadforpastors** [comment]
So I've been fiddling around with this URL scheme and I'm really impressed!
Just a couple things:
- When I use the "Look Up" function on the iPad, it pops up in a mid screen window, instead of a split screen. I would personally prefer this to open in the split screen.
- Especially when using the "bible/" function, but really all of them, I would love for the URL's to work, even without a search term. This would be especially helpful for accessing books other than Bible's, since it would allow me to open any favored resource, such as a daily devotional, from my launcher.
- Here are three of the workflows I've used with the new URL scheme. The first works with Workflow.app, the next two are for the Drafts.app.
     https://workflow.is/workflows/61b8ca580cb84f0486f370fe6b10160f
     https://drafts4-actions.agiletortoise.com/a/1zd
     https://drafts4-actions.agiletortoise.com/a/1ze

**Comment by tomhamming** [comment]
@theipadforpastors I will look into opening the Lookup results in the split window. That does actually make sense. Would it be good to make the split window open into its almost-full-screen mode (what you get from double-tapping the nav bar of the Resource Guide or tapping Full Screen in the gear menu right now)?

Also, what do you mean by having the functions work without a search term? Search needs a product to search (iOS doesn't currently support full-library search, and that's another conversation). If you're looking to be able to open any arbitrary product with a URL, that would probably be a separate URL option, like `olivetree://product/<product ID or name>`.

Lastly, note that you can jump into your default or last-used reading plan with `olivetree://startplan`.

**Comment by theipadforpastors** [comment]
@tomhamming Yeah, opening to the Look Up function in full screen sounds good. By the way, I don't know when you turned on the "double tap for full screen" on the iPad but I love it. It's been there on iPhone for a long time, but it didn't used to work on iPad.
I was able to imitate the function I was referring to by opening a search with just a "space". When you do that, the search panel is open and when you initiate a search, it is with the resource specified by the URL. This is useful because of the autocomplete available in the app that isn't there in Launch Center Pro, obviously.
Also, I am really looking forward to full library search, since I'm going to assume that coming sooner or later (and I know it may be more later than sooner, which is fine. It's, IMO, more of a nice to have than a must. When it's implemented, I'm hoping it takes its cues from the Resource Guide UI. Now THAT would be full library search that would be useful).
Regarding opening any arbitrary resource, yes, I realize that would be a separate URL. I tried Bible/<product ID> but it doesn't work without a location. That's what I'd like changed, so that if the URL olivetree://bible/<product ID or name> is initiated (without a reference included), it will simply open the resource to the last location.
Thanks so much for considering my suggestions.

**Comment by jgclark** [comment]
I've just discovered this repo and this issue.
@tomhamming, I was really looking forward to these working, but for me only the `olivetree://bible/passage[/version]` works.

None of these appear to do anything for me on v6.9.2 on macOS 10.15.7:
```
open "olivetree://lookup/repentance"
open "olivetree://search/door"
open "olivetree://search/door/NIV"
open "olivetree://startplan"
```

BTW, may I suggest it would be helpful to update the documents with all these goodies, including how to lookup the  '<product ID>'. (Only in the library -> book -> right click to 'View resource info' -- and not in the bookstore.)

**Comment by tomhamming** [comment]
@jgclark these URLs are actually meant to work on the iOS app, not the Mac app.

**Comment by RedInkApps** [comment]
And they are still working beautifully on iOS. I love the Olive Tree app.

**Comment by jgclark** [comment]
> @jgclark these URLs are actually meant to work on the iOS app, not the Mac app.

Thanks, @tomhamming, for this detail. The README doesn't say its only for iOS ... though I grant the only examples are iOS. 

Which is a shame, as they'd be just as valuable, surely, for Mac users?  And as I showed above, some of them work, so it would be great to have these ported over to Mac too. I can imagine that most of them would be more easily triggered by Mac launchers and helpers such as Alfred, than on iOS.

**Comment by Pi4JC** [comment]
Any update on what works on the actual mac Version?
